File: code45.py
class Solution(object):
    def jump(self, nums):
        current=0
        jump_cnt=0
        target=len(nums)-1
        while  True:
            if current>=target:
                return jump_cnt
            current_max_jumps=nums[current]
            max_index=0
            next_max_jumps=-1
            for index in range(current+1,current+current_max_jumps+1):
                if index>=target:
                    return jump_cnt+1
                if nums[index]+index>=next_max_jumps:
                    max_index=index
                    next_max_jumps=nums[index]+index
            current=max_index
            jump_cnt+=1
s=Solution()
print(s.jump([10,9,8,7,6,5,4,3,2,1,1,0]))

File: code45.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int jump(vector<int> &nums)
    {
        vector<int64_t> dp(nums.size(), INT_MAX);
        dp[nums.size() - 1] = 0;
        for (int i = nums.size() - 2; i >= 0; i--)
        {
            int64_t m = INT_MAX;
            for (int j = 1; j <= nums[i]; j++)
            {

                if (i + j >= nums.size())
                    break;
                m = min(m, dp[i + j] + 1);
            }
            dp[i] = m;
        }
        return dp[0];
    }
};